tests/data/data_generator_test.py

```diff
@@ -103,6 +103,26 @@ def test_sort_every_batch_actually_adds_noise_every_batch(self):
         second_epoch = [self.as_list(x[0]) for x in second_epoch_arrays]
         assert first_epoch != second_epoch
 
+    def test_padding_is_not_applied_on_padded_instances(self):
+        # Just check that padding lengths which are retrieved for the
+        # instances are the same before and after being put through
+        # the generator. If the dataset was not copied in the generator
+        # we would pad instances and then recalculate the lengths to
+        # pad based off of the padded instances - this checks that doesn't happen.
+        params = Params({
+                'padding_noise': 0.0,
+                'sort_every_epoch': True,
+                })
+        generator = DataGenerator(self.text_trainer, params)
+        # Initial padding lengths
+        dataset = IndexedDataset(self.instances)
+        initial_padding_lengths = [instance.get_padding_lengths() for instance in dataset.instances]
+        batches = generator.create_generator(dataset)
+        # Generate a bit.
+        _ = [next(batches) for _ in range(5)]
```

Comment:
I think a better test would be to generate several epochs worth of batches, and make sure that the lengths of the instances are constant in each batch.